### PR TITLE
feature(signup): implement social media signup

### PR DESCRIPTION
**Pull request by Felglitzzz**
#### What does this PR do?
Ensures user can sign up with their social media accounts

#### Description of Task to be completed?
- Add client id, secret and callback to the environment variables
- Reference the client id, secret and callback url in the passport's social media strategy

#### How should this be manually tested?

- npm run git clone `https://github.com/andela/nazgul-cfh.git` to clone the repository
- Navigate to the nazgul-cfh folder
- Install dependencies with `npm install`
- install and run mongodb
- Create a .env file and add fields to it following the directive in the .env.example file
- run `npm start`
- Navigate to `localhost:3001` and click on sign up
- Click on any of the social media buttons

#### Any background context you want to provide?
Users are requested to validate their login credentials on the selected social platform

#### What are the relevant pivotal tracker stories?
#156449824

### CODE REVIEW

File: public/js/directives.js [hunk 1]
```diff
@@ -58,8 +58,7 @@ angular.module('mean.directives', [])
       templateUrl: '/views/question.html',
       link: function(scope, elem, attr) {}
     };
-  })
-  .directive('timer', function(){
+  }).directive('timer', function(){
```
[the remark this answers is not in the document]
Comment (andela-morekoya) [reply public/js/directives.js]: @iidrees how is it going with excluding lib files like this one from hound checks?

**Comment by andela-morekoya** [comment]
@Felglitzzz won't it be better if there was consistency when users are asked to give authorization?
screen shot 2018-04-06 at 8 52 57 pm
Top mentions the app. Button mentions you

File: public/views/index.html [hunk 2]
```diff
@@ -240,7 +240,7 @@ <h3>XOXO</h3>
 
 <!-- Footer Start -->
 <section class="foot">
-  <div><small class="legal">Legal stuff: This game is based on the <a href="http://cardsagainsthumanity.com">Cards Against Humanity</a> card game, which is CC licensed (BY-NC-SA). Cards for Humanity is not affiliated with Cards Against Humanity. In complying with the Creative Commons license of the Cards Against Humanity card game, all proceeds from donations go directly to charity, as managed by <a href="http://crowdrise.com">crowdrise</a>.</small></div>
+  <div><small class="legal">Legal stuff: This game is based on the <a href="http://localhost:3000">Cards Against Humanity</a> card game, which is CC licensed (BY-NC-SA). Cards for Humanity is not affiliated with Cards Against Humanity. In complying with the Creative Commons license of the Cards Against Humanity card game, all proceeds from donations go directly to charity, as managed by <a href="http://crowdrise.com">crowdrise</a>.</small></div>
```
Comment (andela-morekoya) [inline public/views/index.html]: This is the license so I believe it should actually link to the cardsagainsthumanity site

File: test/user/signup.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      })
+      .end((err, res) => {
+        const { error } = res.body;
+        expect(res.status).to.equal(400);
+        expect(error).to.equal('All fields are required');
+        done();
+      });
+  });
+
+  it('should not allow a user to sign up with no email', (done) => {
```
Comment (andela-morekoya) [inline test/user/signup.js]: What's the difference between this and the previous test?

File: test/user/signup.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      })
+      .end((err, res) => {
+        const { error } = res.body;
+        expect(res.status).to.equal(400);
+        expect(error).to.equal('All fields are required');
+        done();
+      });
+  });
+
+  it('should not allow a user to sign up with no password', (done) => {
```
Comment (andela-morekoya) [inline test/user/signup.js]: Another repeated test?

File: test/user/signup.js (hunk 3 above)
Comment (Felglitzzz) [reply test/user/signup.js]: The previous test covers the model. This test covers validation and edge cases.

File: test/user/signup.js (hunk 4 above)
Comment (Felglitzzz) [reply test/user/signup.js]: This test covers for each missing fields.

File: public/views/index.html (hunk 2 above)
Comment (Felglitzzz) [reply public/views/index.html]: I have implemented the feedback
